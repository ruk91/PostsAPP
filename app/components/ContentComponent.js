import React, {Fragment} from 'react';
import { 
    View, 
    Text, 
    Dimensions,
    StyleSheet
} from 'react-native';

const window = Dimensions.get('window');

class ContentComponent extends React.Component {
    render () {
        let content = this.props.content;

        return (
            <View style={styles.postContent}>
                <Text style={styles.postTitle}>
                    {content}
                </Text>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    postContent: {
        flex: 1,
        padding:30,
      },
      postTitle:{
        fontSize:26,
        fontFamily: 'Nunito-Regular',
        fontWeight:'600',
      },
})

export default ContentComponent;