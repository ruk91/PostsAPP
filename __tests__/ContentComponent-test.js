import React, {Fragment} from 'react';
import { 
    View, 
    Text, 
    TouchableOpacity, 
    Dimensions,
    StyleSheet
} from 'react-native';
// import { expect } from 'chai';
import { expect as chaiExpect } from 'chai'
import Enzyme from 'enzyme';
import { shallow, configure, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
const window = Dimensions.get('window');
import ContentComponent from '../app/components/ContentComponent';

configure({adapter: new Adapter()});

describe('<ContentComponent />', () => {

    it('should render the ContentComponent Component', () => {
        const wrapper = shallow(<ContentComponent />);
        chaiExpect(wrapper.find(ContentComponent)).to.have.lengthOf(1);
    });

    it('renders one <View /> components', () => {
        const wrapper = shallow(<ContentComponent />);
        chaiExpect(wrapper.find(View)).to.have.lengthOf(1);
    });

    it('renders one <Text /> components', () => {
        const wrapper = shallow(<ContentComponent />);
        chaiExpect(wrapper.find(Text)).to.have.lengthOf(1);
    });

});

describe('Styling in <ContentComponent/>', () => {

    it('renders ContentComponent.postContent styles correctly', () => {
        const tree = renderer.create(
            <ContentComponent>
                <ContentComponent.postContent 
                    flex= {1}
                    padding= {30}
                />
            </ContentComponent>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });


    it('renders ContentComponent.postTitle styles correctly', () => {
        const tree = renderer.create(
            <ContentComponent>
                <ContentComponent.postTitle 
                    fontSize= {26}
                    fontFamily= 'Nunito-Regular'
                    fontWeight = '600'
                />
            </ContentComponent>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

});