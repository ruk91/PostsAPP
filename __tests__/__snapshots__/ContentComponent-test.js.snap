// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styling in <ContentComponent/> renders ContentComponent.postContent styles correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 30,
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Nunito-Regular",
        "fontSize": 26,
        "fontWeight": "600",
      }
    }
  />
</View>
`;

exports[`Styling in <ContentComponent/> renders ContentComponent.postTitle styles correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 30,
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Nunito-Regular",
        "fontSize": 26,
        "fontWeight": "600",
      }
    }
  />
</View>
`;
